gn2/wqflask/views.py

```diff
@@ -1490,3 +1492,48 @@ def __success__(results):
         return redirect(url_for("view_diff",
                                 inbredset_id=inbredset_id,
                                 diff_id=form["diff_id"]))
+
+
+@app.route("/wiki/<int:comment_id>/edit", methods=["GET", "POST"])
+def edit_wiki(comment_id: int):
+    """fetch generif metadata from gn3 and display it"""
+    # FIXME: better error handling
+    if request.method == "GET":
+        last_wiki_content = (
+            monad_requests.get(urljoin(GN3_LOCAL_URL, f"/api/metadata/wiki/{comment_id}"))
+            .either(lambda err: err.raise_for_status(), lambda x: x.json())
+        )
+
+        species_dict = (
+            monad_requests.get(urljoin(GN3_LOCAL_URL, "/api/metadata/wiki/species"))
+            .either(lambda err: err.raise_for_status(), lambda x: x.json())
+        )
+        categories = (
+            monad_requests.get(urljoin(GN3_LOCAL_URL, "/api/metadata/wiki/categories"))
+            .either(lambda err: err.raise_for_status(), lambda x: list(x.json().keys()))
+        )
```

Comment:
*thought: question:* I'm not a big fan of monad requests.  Is there a good reason for using it?

Reply:
*suggestion:* Monad requests makes things un-necessarily complex.  Please remove.

Reply:
Thanks. I've dropped this and used the `requests` library instead. I've also realized the module may be buggy since it treats any other 2XX requests as errors except 200.